pkg/sidecar_mounter/sidecar_mounter.go

```diff
@@ -224,14 +225,18 @@ func collectMetrics(ctx context.Context, port, dirPath string) {
 			return
 		case <-ticker.C:
 			newCtx, cancel := context.WithTimeout(ctx, 5*time.Second)
-			scrapeMetrics(newCtx, metricEndpoint, outputPath)
+			scrapeMetrics(newCtx, metricEndpoint, dirPath, genNumber)
+			deleteOutdatedFile(dirPath, genNumber-2)
```

Comment:
What is the reasoning behind "-2"?

Reply:
If x is the current generation, we are leaving some buffer where the outdated (x-1) file exists so that if this file is being read it has time to finish. The -2 means we are deleting file generations up to x - 2.

Reply:
Can you add a comment with the reasoning?

Reply:
Sure! Pending on comment above regarding keeping this approach.

Reply:
Done.